src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -282,12 +282,12 @@ static int GetHashCodeWorker(object? o)
                 return false;
 
             // Compare the contents
-            return ContentEquals(o1, o2);
+            return SpanHelpers.SequenceEqual(
```

Comment:
This introduces reliability problem. The unmanaged implementation of ContentEquals depends on the GC not moving objects around while the comparison is in progress. It is not the case for `SpanHelpers.SequenceEqual`. SpanHelpers.SequenceEqual is managed and so the following is possible:
1. Object reference read from the first object
2. GC kicks on and moves the objects around
3. Object reference read from the second object
4. This API returns false incorrectly.

Reply:
This is legacy API that is really only used by VB in some cases, and this path is only used for valuetypes that are expected to be small that means the API is not going to be long running. It should be ok to just delete the GC poll.